### PR TITLE
Add the capability to record a batch of measurements.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: api/src/main/java/io/opentelemetry/metrics/Meter.java [hunk 1]
```diff
@@ -173,4 +173,13 @@
...
   MeasureLong.Builder measureLongBuilder(String name);
+
+  /**
+   * Utility method that allows users to atomically record measurements to a set of Measures.
+   *
+   * @return a {@code MeasureBatchRecorder} that can be use to atomically record a set of
+   *     measurements associated with different Measures.
+   * @since 0.1.0
+   */
+  MeasureBatchRecorder recordMeasureBatch();
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/Meter.java]: Suggestion:
```suggestion
  MeasureBatchRecorder newMeasureBatchRecorder();
```

File: api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+ * <p>This class is equivalent with individually calling record on every Measure, but has the
+ * advantage that all these operations are recorded atomically and it is more efficient.
+ */
+public interface MeasureBatchRecorder {
+  /**
+   * Associates the {@link MeasureLong} with the given value. Subsequent updates to the same {@link
+   * MeasureLong} will overwrite the previous value.
+   *
+   * @param measure the {@link MeasureLong}.
+   * @param value the value to be associated with {@code measure}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java]: Also worth mentioning values cannot be negative.

File: api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+   * @since 0.1.0
+   */
+  MeasureBatchRecorder put(MeasureLong measure, long value);
+
+  /**
+   * Associates the {@link MeasureDouble} with the given value. Subsequent updates to the same
+   * {@link MeasureDouble} will overwrite the previous value.
+   *
+   * @param measure the {@link MeasureDouble}.
+   * @param value the value to be associated with {@code measure}.
```
Comment (songy23) [inline api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java]: Same here

File: api/src/test/java/io/opentelemetry/metrics/MeasureBatchRecorderTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link MeasureBatchRecorder}. */
+@RunWith(JUnit4.class)
+public class MeasureBatchRecorderTest {
```
Comment (songy23) [inline api/src/test/java/io/opentelemetry/metrics/MeasureBatchRecorderTest.java]: Nit: add tests on negative values.

File: api/src/main/java/io/opentelemetry/metrics/Meter.java (hunk 1 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/Meter.java]: Done.

File: api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java (hunk 2 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java]: Done. Via the throws statement.

File: api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java (hunk 3 above)
Comment (bogdandrutu) [reply api/src/main/java/io/opentelemetry/metrics/MeasureBatchRecorder.java]: Done. Via the throws statement.

File: api/src/test/java/io/opentelemetry/metrics/MeasureBatchRecorderTest.java (hunk 4 above)
Comment (bogdandrutu) [reply api/src/test/java/io/opentelemetry/metrics/MeasureBatchRecorderTest.java]: Done.

**Pull request merged**